system_baseline/views/v1.py

```diff
@@ -278,6 +280,42 @@ def create_baseline(system_baseline_in):
     return baseline.to_json()
 
 
+@metrics.baseline_create_requests.time()
+@metrics.api_exceptions.count_exceptions()
+def create_baseline_from_hsp(hsp_id, display_name):
+    """
+    create a baseline from an hsp
+    """
+    account_number = view_helpers.get_account_number(request)
+
+    _validate_uuids([hsp_id])
+
+    # ensure display_name is not null
+    if not display_name:
+        raise HTTPError(
+            HTTPStatus.BAD_REQUEST, message="no value given for display_name"
+        )
+
+    account_number = view_helpers.get_account_number(request)
+    _check_for_existing_display_name(display_name, account_number)
+    _check_for_whitespace_in_display_name(display_name)
+
+    query = HistoricalSystemProfile.query.filter(
```

Comment:
You'll need to call the HSP service here instead of querying the DB.

I think you can use `from kerlescan.hsp_service_interface import fetch_historical_sys_profiles` similar to what's used in https://github.com/RedHatInsights/drift-backend/blob/master/drift/views/v1.py#L153-L158